orbit/pkg/update/runner.go

```diff
@@ -140,7 +144,7 @@ func (r *Runner) UpdateAction() (bool, error) {
 		}
 		// Check whether the hash of the repository is different than
 		// that of the target local file.
-		if !bytes.Equal(r.localHashes[target], metaHash) {
+		if !bytes.Equal(r.localHashes[target], metaHash) || needsSymlinkUpdate {
```

Comment:
`needsSymlinkUpdate ` is a special case for the `orbit` target only and this loop loops through all registered targets (`orbit`, `osqueryd`, `fleet-desktop` and extensions, etc.) so we should run `needsSymlinkUpdate` inside the loop:
```go
  needsSymlinkUpdate := false
  if target == "orbit" {
    var err error
    needsSymlinkUpdate, err = r.needsSymlinkUpdate()
    if err != nil {
      return false, fmt.Errorf("check symlink failed: %w", err)
    }
  }
```

Reply:
Should be fixed now